src/dashboard/src/components/administration/views.py

```diff
@@ -450,15 +451,17 @@ def general(request):
         'False': False}.get(
             initial_data.get('storage_service_use_default_config', True),
             True)
+    general_form = GeneralSettingsForm(request.POST or None,
+                                       prefix='general', initial=initial_data)
     storage_form = StorageSettingsForm(request.POST or None,
                                        prefix='storage', initial=initial_data)
     checksum_form = ChecksumSettingsForm(request.POST or None,
                                          prefix='checksum algorithm',
                                          initial=initial_data)
 
-    if storage_form.is_valid() and checksum_form.is_valid():
-        storage_form.save()
-        checksum_form.save()
+    forms = (general_form, storage_form, checksum_form)
+    if map(lambda form: form.is_valid(), forms):
```

Comment:
I'm not sure this condition is correct. No matter how many of the forms are valid or invalid, the return value of `map` is still going to be something that is `True` when `bool` is applied to it.